src/Tools/ExternalAccess/FSharp/Classification/IFSharpClassificationService.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.FSharp.Classification
+{
+    internal interface IFSharpClassificationService
```

Comment:
💡  This is interesting because of two things:

1. It's a duplicate of `IClassificationService` (sometimes duplication will happen in the IVT adapter assemblies)
2. None of the members have internal types in their signatures

The second item means the *only* visibility constraint leading to this adapter interface is the need for `typeof(IClassificationService)` in the `[ExportLanguageService]` line above. An alternative solution would be the following:

1. Modify `FSharpClassificationService` to be `abstract`, and make all methods in it which implement `IClassificationService` abstract
2. Instead of exporting `FSharpClassificationService`, export an `FSharpClassificationServiceFactory` which imports the concrete instance of `FSharpClassification` and returns it

Since this change would be a rather substantial simplification, it should be treated like ❕ unless the strategy turns out to not be viable for some reason.

Reply:
I dont get it. FSharp still need to export that we need to import. FSharp still need IVT to Workspace once they start to move thier other IVT access to this dll.

so I am not sure what they are buying?

Reply:
The F# implementation would export `FSharpClassificationService`, which would be defined in this assembly. The implementation of `FSharpClassificationServiceFactory` would be provided by this assembly.

Adapter code:

```csharp
internal abstract class FSharpClassificationService : IClassificationService
{
  // abstract methods here to implement IClassificationService
}

[ExportLanguageServiceFactory(...)]
internal sealed class FSharpClassificationServiceFactory : ILanguageServiceFactory
{
  public FSharpClassificationServiceFactory(FSharpClassificationService classificationService)
  {
    // store classificationService so it can be returned from the factory when needed
  }
}
```

FSharp implementation code:

```csharp
[Export(typeof(FSharpClassificationService))]
internal sealed class FSharpClassificationServiceImpl : FSharpClassificationService
{
  // Implement abstrat methods from FSharpClassificationService
}
```